### PR TITLE
[Storage] Migrate `test_file_system.py` to test proxy 

### PR DESCRIPTION
**Pull request by vincenttran-msft**
This PR migrates `test_file_system.py` and `test_file_system_async.py` to test proxy. This PR also exposes the `softdel` account for DataLake and so ensure live pipeline passes against this PR!!

- [x] Note: When CI passes, run live test against this PR!

### CODE REVIEW

File: sdk/storage/azure-storage-file-datalake/tests/test_file_system.py [hunk 1]
```diff
@@ -416,41 +489,46 @@ def test_list_system_filesystems(self, datalake_storage_account_name, datalake_s
...
+        with pytest.raises(HttpResponseError):
             self.dsc._rename_file_system(name=old_name, new_name=new_name)
-        with self.assertRaises(HttpResponseError):
+        with pytest.raises(HttpResponseError):
             self.dsc._rename_file_system(name=old_name, new_name=new_name, lease="bad_id")
         new_filesystem = self.dsc._rename_file_system(name=old_name, new_name=new_name, lease=filesystem_lease_id)
-        self.assertEqual(new_name, new_filesystem.get_file_system_properties().name)
+        assert new_name == new_filesystem.get_file_system_properties().name
 
+    @pytest.mark.playback_test_only
```
Comment (jalauzon-msft) [inline sdk/storage/azure-storage-file-datalake/tests/test_file_system.py:513]: With using the soft delete account from the ARM template, this _should_ be able to run Live now. Same with all soft delete tests.

File: sdk/storage/azure-storage-file-datalake/tests/test_file_system_async.py [hunk 2]
```diff
@@ -1,20 +1,16 @@
-# coding: utf-8
-
 # -------------------------------------------------------------------------
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License. See License.txt in the project root for
 # license information.
 # --------------------------------------------------------------------------
 import asyncio
+
 import pytest
```
Comment (jalauzon-msft) [inline sdk/storage/azure-storage-file-datalake/tests/test_file_system_async.py:8]: Move `pytest` down on top of `azure.core` (not built-in) and remove new line to join `asyncio` and `unittest`.

**Review by jalauzon-msft — COMMENTED** [review]
Looks good! Just a couple minor things (hopefully).

File: sdk/storage/azure-storage-file-datalake/tests/test_file_system_async.py (hunk 2 above)
Comment (vincenttran-msft) [reply sdk/storage/azure-storage-file-datalake/tests/test_file_system_async.py:8]: Doh. This one always gets me. Will fix 😄

**Review by jalauzon-msft — APPROVED** [review]
Looks good, thanks! Let's be sure to run Live tests against this before merging to be sure the datalake soft deleted account works there.

**Pull request merged**